Mathlib/Order/WithBot.lean

```diff
@@ -829,6 +829,14 @@ lemma forall_coe_le_iff_le [NoMaxOrder α] {x y : WithTop α} : (∀ a : α, a 
 
 end Preorder
 
+section PartialOrder
+variable [PartialOrder α] [NoMaxOrder α] {x y : WithTop α}
+
+lemma eq_of_forall_coe_le_iff (h : ∀ a : α, a ≤ x ↔ a ≤ y) : x = y :=
+  le_antisymm (forall_coe_le_iff_le.mp fun a ↦ (h a).1) (forall_coe_le_iff_le.mp fun a ↦ (h a).2)
```

Comment:
Can we have the `WithBot` version too?

Reply:
Done. The preceding lemma `forall_coe_le_iff_le` didn't have a `WithBot` version either, so I made that too, and deduced the WithTop versions using the usual trick of applying the WithBot version to the order dual.